can/interfaces/ixxat/canlib.py

```diff
@@ -495,24 +505,32 @@ def send(self, msg, timeout=None):
             adapter = (ctypes.c_uint8 * len(msg.data)).from_buffer(msg.data)
             ctypes.memmove(message.abData, adapter, len(msg.data))
 
-        if timeout:
-            _canlib.canChannelSendMessage(
-                self._channel_handle, int(timeout * 1000), message)
-        else:
-            _canlib.canChannelPostMessage(self._channel_handle, message)
+        try:
+            if timeout:
+                _canlib.canChannelSendMessage(self._channel_handle, int(timeout * 1000), message)
+            else:
+                _canlib.canChannelPostMessage(self._channel_handle, message)
+        except VCITimeout:
+            # if the user wanted an timeout, the timeout in the library is probably no error
+            if timeout:
+                return False
+            else:
+                raise(CanOperationError())
```

Comment:
no brackets needed here:

```suggestion
                raise CanOperationError("") # TODO: give some reason
```

Same for below.

Reply:
Ok. I'll add some text. This will be more or less specific. A point for later improvements.